aten/src/ATen/native/Convolution.cpp

```diff
@@ -829,7 +844,13 @@ std::tuple<Tensor,Tensor,Tensor> _convolution_double_backward(
   params.dilation = dilation_.vec();
   params.transposed = transposed_;
   params.output_padding = output_padding_.vec();
-  params.groups = groups_;
+  // TODO: hacky way of inferring the groups number for grouped Conv3D
+  // See: https://github.com/pytorch/pytorch/pull/36355
+  if (!params.transposed && input.dim() > 4) {
+    params.groups = input.size(1) / weight.size(1);
```

Comment:
So potentially this could mask the issue of someone putting in input with incompatible sizes? E.g. I have Conv3d(16,32) that expects input with 16 channels. I send an input with 32 channels by mistake, and what'll happen here is instead of erroring out you say - ok, we'll run convolution with 2 groups. This seems unsafe?

Reply:
The shape will be checked in the forward pass here:
https://github.com/pytorch/pytorch/blob/ee3d046f8783e9a0a30305bc9ad264dc48061ecf/aten/src/ATen/native/Convolution.cpp#L584
As long as the user doesn't call double_backward directly we should be fine, right? We are counting on that the backward call is called by autograd thus removing the user error. Otherwise, we need to modify tools/autograd/derivatives.yaml and specify the group parameter.

Reply:
Thanks for clarifying! That's ok then, user cannot call double backward directly (it's not exposed), so we should be fine.